lightning/src/offers/contacts.rs

```diff
@@ -0,0 +1,366 @@
+// This file is Copyright its original authors, visible in version control
+// history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+//! Data structures and utilities for managing Lightning Network contacts.
+//!
+//! Contacts are trusted people to which we may want to reveal our identity when paying them.
+//! We're also able to figure out when incoming payments have been made by one of our contacts.
+//! See [bLIP 42](https://github.com/lightning/blips/blob/master/blip-0042.md) for more details.
+
+use crate::blinded_path::IntroductionNode;
+use crate::offers::offer::Offer;
+use bitcoin::hashes::{sha256, Hash, HashEngine};
+use bitcoin::secp256k1::Scalar;
+use bitcoin::secp256k1::{PublicKey, Secp256k1, SecretKey};
+use core::fmt;
+
+#[allow(unused_imports)]
+use crate::prelude::*;
+
+/// BIP 353 human-readable address of a contact.
+///
+/// This represents an address in the form `name@domain` that can be used to identify
```

Comment:
We already have `lightning::onion_message::dns_resolution::HumanReadableName`, let's not duplicate it?